xc/xc7/tests/soc/litex/generate_linux.py

```diff
@@ -0,0 +1,227 @@
+            soc_kwargs.update(uart_name="usb_acm")
+        if "ethernet" in board.soc_capabilities:
+            soc_kwargs.update(with_ethernet=True)
+
+        # SoC creation -----------------------------------------------------------------------------
+        soc = SoCLinux(board.soc_cls, **soc_kwargs)
+        board.platform = soc.platform
+
+        # SoC peripherals --------------------------------------------------------------------------
+        if board_name in ["arty", "arty_a7"]:
+            from litex_boards.platforms.arty import _sdcard_pmod_io
+            board.platform.add_extension(_sdcard_pmod_io)
+
+        if "mmcm" in board.soc_capabilities:
+            soc.add_mmcm(2)
+        if "spiflash" in board.soc_capabilities:
+            soc.add_spi_flash(dummy_cycles=board.SPIFLASH_DUMMY_CYCLES)
+            soc.add_constant("SPIFLASH_PAGE_SIZE", board.SPIFLASH_PAGE_SIZE)
+            soc.add_constant(
+                "SPIFLASH_SECTOR_SIZE", board.SPIFLASH_SECTOR_SIZE
+            )
+        if "spisdcard" in board.soc_capabilities:
+            soc.add_spi_sdcard()
+        if "sdcard" in board.soc_capabilities:
+            soc.add_sdcard()
+        if "ethernet" in board.soc_capabilities:
+            soc.configure_ethernet(
+                local_ip=args.local_ip, remote_ip=args.remote_ip
+            )
+        # if "leds" in board.soc_capabilities:
+        # soc.add_leds()
+        if "rgb_led" in board.soc_capabilities:
+            soc.add_rgb_led()
+        if "switches" in board.soc_capabilities:
+            soc.add_switches()
+        if "spi" in board.soc_capabilities:
+            soc.add_spi(args.spi_data_width, args.spi_clk_freq)
+        if "i2c" in board.soc_capabilities:
+            soc.add_i2c()
+        if "xadc" in board.soc_capabilities:
+            soc.add_xadc()
+        if "framebuffer" in board.soc_capabilities:
+            assert args.video in video_resolutions.keys(
+            ), "Unsupported video resolution"
+            video_settings = video_resolutions[args.video]
+            soc.add_framebuffer(video_settings)
+        if "icap_bitstream" in board.soc_capabilities:
+            soc.add_icap_bitstream()
+        soc.configure_boot()
+
+        # Build ------------------------------------------------------------------------------------
+        build_dir = args.builddir
+        builder = Builder(
+            soc,
+            output_dir=build_dir,
+            csr_json=os.path.join(build_dir, "csr.json"),
+            bios_options=["TERM_MINI"]
+        )
+        builder.build(build_name="top", run=args.build)
+
+        # DTS --------------------------------------------------------------------------------------
+        # soc.generate_dts(build_dir, board_name)
+        # soc.compile_dts(build_dir, board_name)
+
+        # Machine Mode Emulator --------------------------------------------------------------------
+        # soc.compile_emulator(build_dir)
```

Comment:
Should probably add a comment why this is disabled

Reply:
Done.